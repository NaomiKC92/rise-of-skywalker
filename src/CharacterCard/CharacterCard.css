* {
  box-sizing: border-box;
  font-family: 'PT Sans Narrow', sans-serif;
}

.character-card {
  background-color: rgb(160, 165, 168);
  border: 1px solid #fff;
  text-align: center;
  display: flex;
  flex-direction: column;
  font-size: 21px;
  width: 100%;
  margin: 15px;
  padding: 1.5rem;
  color: #fff;
  border-radius: 25px;
}

.favorite-image {
  cursor: pointer;
  transition: opacity 0.2s ease-in;
  width: 50px;
  color: #fff;
}

p {
  padding-top: 10px;
  padding-bottom: 10px;
}

.episode-list {
  min-height: 300px;
  list-style: none;
  padding-top: 20px;
}

.character-name {
  height: 100px;
}

.population {
  height: 100px;
}

.character-card-button {
  background: #fff;
  border: 2px solid #fff;
  border-radius: 3px;
  color: #04243d;
  cursor: pointer;
  display: inline-block;
  font-size: 1.2rem;
  padding: 0.4rem 1.3rem;
  transition: opacity 0.2s ease-in;
  font-weight: bold;
}

.character-card-button:focus {
  outline: none;
}
