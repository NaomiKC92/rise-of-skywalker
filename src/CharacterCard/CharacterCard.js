import React from 'react';
import './CharacterCard.css';
import isFavoriteFalse from '../../src/images/isFavoriteFalse.svg';
import isFavoriteTrue from '../../src/images/isFavoriteTrue.svg';

const CharacterCard = ({ character, setupFavoriteCharacters }) => {
  console.log('Character Card:', character);
  const episodeApperances = character.films.map((film, i) => (
    <li key={i}>{film}</li>
  ));
  return (
    <div>
      <div className='card'>
        <img
          className={'favorite-image'}
          src={isFavoriteFalse}
          alt='favorite icon'
        />
        <h2>Name: {character.name}</h2>
        <p>Homeworld: {character.homeworld}</p>
        <p>Population: {character.population}</p>
        <p>Species: {character.species}</p>
        <ul className='episode-list'>Appear in: {episodeApperances}</ul>
        <button
          onClick={() =>
            setupFavoriteCharacters({ ...character }, character.name)
          }
          className='button'
          type='button'>
          Favorite
        </button>
      </div>
    </div>
  );
};

export default CharacterCard;
