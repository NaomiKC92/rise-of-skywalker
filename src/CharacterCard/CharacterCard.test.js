import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard';

describe('CHARACTER CARD', () => {
  let wrapper;

  const mockCharacter = {
    homeworld: 'SpaceX Occupying Mars',
    species: 'Human',
    films: ['The Day After Tomorrow', 'The Martian', 'Gravity'],
    name: 'BB8'
  };
  const mockSetupFavoriteCharacters = jest.fn();
  const mockFavoriteCharactersMap = [];

  beforeEach(() => {
    wrapper = shallow(
      <CharacterCard
        character={mockCharacter}
        setupFavoriteCharacters={mockSetupFavoriteCharacters}
        favoriteCharactersMap={mockFavoriteCharactersMap}
      />
    );
  });

  it('should match the CharacterCard Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
