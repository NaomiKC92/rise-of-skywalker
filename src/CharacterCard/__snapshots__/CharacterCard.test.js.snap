// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHARACTER CARD should match the CharacterCard Component Snapshot 1`] = `
<div>
  <div
    className="character-card"
  >
    <img
      alt="favorite icon"
      className="favorite-image"
      onClick={[Function]}
      src="isFavoriteFalse.svg"
    />
    <h2>
      BB8
    </h2>
    <p>
      Homeworld: 
      SpaceX Occupying Mars
    </p>
    <p>
      Population: 
    </p>
    <p>
      Species: 
      Human
    </p>
    <ul
      className="episode-list"
    >
      Appears in: 
      <li
        key="0"
      >
        The Day After Tomorrow
      </li>
      <li
        key="1"
      >
        The Martian
      </li>
      <li
        key="2"
      >
        Gravity
      </li>
    </ul>
    <button
      className="character-card-button"
      onClick={[Function]}
      type="button"
    >
      Favorite
    </button>
  </div>
</div>
`;
